### ISSUE TITLE
add toluene as solvent

### ISSUE THREAD

**Issue by orbeckst** [Becksteinlab/MDPOW#222]
For SAMPL9, we want to evaluate solvation free energies in [toluene](https://pubchem.ncbi.nlm.nih.gov/compound/1140).

![Toluene_300](https://user-images.githubusercontent.com/237980/204353673-4291f2e1-df09-4f90-aadb-0ca6ae3c266b.png)

- [x] add force field files AMBER GAFF
- [ ] add force field files CGENFF
- [x] add force field files OPLS-AA (mol2ff)
- [ ] integrate in the solvent selections 
- [ ] add basic tests

In general, follow what we did for cyclohexane.

**Comment by orbeckst** [comment]
GAFF and OPLSAA files on branch [toluene_box](https://github.com/Becksteinlab/MDPOW/tree/toluene_box).

**Comment by VOD555** [comment]
Sure, I'm working on add GAFF and OPLSAA toluene.

**Comment by orbeckst** [comment]
@VOD555 could you please generate the PR to go along with the toluene_box branch? Clicking on https://github.com/Becksteinlab/MDPOW/compare/develop...toluene_box should do it. Thanks!

**Comment by orbeckst** [comment]
Without a PR I have a hard time reviewing/commenting.

**Comment by iorga** [comment]
@orbeckst @VOD555 
The GAFF simulations are now finished, currently being synchronized into `/nfs/homes5/Projects/SAMPL9/SAMPL9-logP`. The compounds S01 to S14 are already copied and the permissions fixed to give write access to all the `sampl` group in the Beckstein lab.

I'm trying to analyze the simulations and I'm facing two problems:
1. The script `mdpow-ptw` is present in the `scripts` folder of the installer, but it's not present in the `setup.py`, so it is not installed with the command `python setup.py install`. @VOD555 could you please add it to `setup.py` ? Thanks.
2. In these conditions, I am using directly the `mdpow-ptw` from the repository, but I get the following error:
```
(/sge/packs/anaconda3_centos6_envs/mdpow_py3) /gem/iorga/repositories/MDPOW/scripts/mdpow-ptw  --estimator alchemlyb /ml/bogdan/2023-SAMPL9-logP/03_sims/GAFF/S01/S01
mdpow       : INFO     MDPOW 0+unknown starting.
mdpow       : INFO     Copyright (c) 2010-2021 Shujie Fan, Ian Kenney, Alia Lescoulie, Bogdan Iorga, and Oliver Beckstein
mdpow       : INFO     Released under the GNU Public Licence, version 3.
mdpow       : INFO     For bug reports and help: https://github.com/Becksteinlab/MDPOW/issues
mdpow.config: INFO     Using the bundled force fields from GMXLIB='/sge/packs/anaconda3_centos6_envs/mdpow_py3/lib/python3.9/site-packages/mdpow/top'.
mdpow.config: INFO     If required, override this behaviour by setting the environment variable GMXLIB yourself.
mdpow       : INFO     Analyzing directory '/ml/bogdan/2023-SAMPL9-logP/03_sims/GAFF/S01/S01'... (can take a while)
mdpow       : INFO     Reading water data '/ml/bogdan/2023-SAMPL9-logP/03_sims/GAFF/S01/S01/FEP/water/Ghyd.fep'
mdpow.fep   : INFO     Solvation free energy calculation for molecule S01 in solvent water.
mdpow.fep   : INFO     Base directory is '/ml/bogdan/2023-SAMPL9-logP/03_sims/GAFF/S01/S01'
mdpow.fep   : INFO     Using setup directories under 'FEP/water': {'coulomb': 'FEP/water/Coulomb', 'vdw': 'FEP/water/VDW'}
mdpow.fep   : INFO     Default checkpoint file is '/ml/bogdan/2023-SAMPL9-logP/03_sims/GAFF/S01/S01/FEP/water/Ghyd.fep'
mdpow       : INFO     Reading toluene data '/ml/bogdan/2023-SAMPL9-logP/03_sims/GAFF/S01/S01/FEP/toluene/Gtol.fep'
mdpow       : CRITICAL Running analysis in directory '/ml/bogdan/2023-SAMPL9-logP/03_sims/GAFF/S01/S01' failed
mdpow       : ERROR    Catastrophic problem occurred, see the stack trace for hints.
Traceback (most recent call last):
  File "/gem/iorga/repositories/MDPOW/scripts/mdpow-ptw", line 258, in <module>
    run_ptw(directory, plotfile=opts.plotfile,
  File "/gem/iorga/repositories/MDPOW/scripts/mdpow-ptw", line 99, in run_ptw
    gwat, gtol = load_gsolv(directory, permissive=kwargs.pop('permissive',False),)
  File "/gem/iorga/repositories/MDPOW/scripts/mdpow-ptw", line 59, in load_gsolv
    gtol = mdpow.fep.Gtol(filename=files['toluene'], basedir=directory, permissive=permissive)
AttributeError: module 'mdpow.fep' has no attribute 'Gtol'
Traceback (most recent call last):
  File "/gem/iorga/repositories/MDPOW/scripts/mdpow-ptw", line 258, in <module>
    run_ptw(directory, plotfile=opts.plotfile,
  File "/gem/iorga/repositories/MDPOW/scripts/mdpow-ptw", line 99, in run_ptw
    gwat, gtol = load_gsolv(directory, permissive=kwargs.pop('permissive',False),)
  File "/gem/iorga/repositories/MDPOW/scripts/mdpow-ptw", line 59, in load_gsolv
    gtol = mdpow.fep.Gtol(filename=files['toluene'], basedir=directory, permissive=permissive)
AttributeError: module 'mdpow.fep' has no attribute 'Gtol'
```
Do you have any idea about what's wrong ?
3. `(/sge/packs/anaconda3_centos6_envs/mdpow_py3) mdpow-solvationenergy -S water  --estimator alchemlyb /ml/bogdan/2023-SAMPL9-logP/03_sims/GAFF/S01/S01` runs without any error.
4. But I get an error for toluene:
```
(/sge/packs/anaconda3_centos6_envs/mdpow_py3) mdpow-solvationenergy -S toluene  --estimator alchemlyb /ml/bogdan/2023-SAMPL9-logP/03_sims/GAFF/S01/S01
mdpow       : INFO     MDPOW 0.8.0+11.ge5ce7af.dirty starting.
mdpow       : INFO     Copyright (c) 2010-2021 Shujie Fan, Ian Kenney, Alia Lescoulie, Bogdan Iorga, and Oliver Beckstein
mdpow       : INFO     Released under the GNU Public Licence, version 3.
mdpow       : INFO     For bug reports and help: https://github.com/Becksteinlab/MDPOW/issues
mdpow.config: INFO     Using the bundled force fields from GMXLIB='/home/chem/iorga/.cache/Python-Eggs/MDPOW-0.8.0+11.ge5ce7af.dirty-py3.9.egg-tmp/mdpow/top'.
mdpow.config: INFO     If required, override this behaviour by setting the environment variable GMXLIB yourself.
usage: mdpow-solvationenergy [options] DIRECTORY [DIRECTORY ...]

Run the free energy analysis for a solvent  in <DIRECTORY>/FEP
and return DeltaG.

DIRECTORY should contain all the files resulting from running
``mdpow.fep.Ghyd.setup()`` (or the corresponding ``Goct.setup()`` or
``Gcyclohexane.setup()`` and the results of the MD FEP
simulations. It relies on the canonical naming scheme (basically: just
use the defaults as in the tutorial).

The dV/dlambda plots can be produced automatically (--plot=auto). If multiple
DIRECTORY arguments are provided then one has to choose the auto option (or
None).

The total solvation free energy is calculated as

  DeltaG* = -(DeltaG_coul + DeltaG_vdw)

Note that the standard state refers to the "Ben-Naim" standard state
of transferring 1 M of compound in the gas phase to 1 M in the aqueous
phase.

Results are *appended* to a data file with **Output file format**::

          .                 ---------- kJ/mol ---
          molecule solvent  DeltaG*  coulomb  vdw

All observables are quoted with an error estimate, which is derived
from the correlation time and error propagation through Simpson's rule
(see :meth:`mdpow.fep.Gsolv`). It ultimately comes from the error on
<dV/dlambda>, which is estimated as the error to determine the
average.

molecule
    molecule name as used in the itp
DeltaG*
    solvation free energy vacuum --> solvent, in kJ/mol
coulomb
    discharging contribution to the DeltaG*
vdw
    decoupling contribution to the DeltaG*
directory
    folder in which the simulations were stored
mdpow-solvationenergy: error: argument --solvent/-S: invalid choice: 'toluene' (choose from 'water', 'octanol', 'cyclohexane')
```
Could you please have a look into these errors please? More generally, toluene should be implemented in all scripts from the `scripts` folder, which is currently not the case.

**Comment by VOD555** [comment]
@iorga 
1. Sure. I'm working on this.
2. After I fixed 1, and reinstall MDPOW, I didn't get these errors when `mdpow-ptw  --estimator S01`. Maybe it's due to 1
4. We didn't add options in this script. I'm adding options, and will use S01 to test locally.

**Comment by iorga** [comment]
Thanks @VOD555 . I tested and both `mdpow-ptw` and `mdpow-solvationenergy -S toluene` work now. I think we can include this change in the develop branch.

There are still some scripts (e.g. `mdpow-rebuild-fep`) that do not have implemented the new solvents (cyclohexane, toluene). This is of lower priority right now, but should be done before the new release, to have a homogeneous package. I will open a separate issue for it.

Now I am running the analysis for all GAFF sims and will rsync the results when it will be finished.
